merger.go

```diff
@@ -493,9 +486,9 @@ func (m *Merger) mergeUpload() (uploaded []string, err error) {
 
 	zlog.Info("merged and uploaded", zap.String("filename", fileNameForBlocksBundle(b.lowerBlock)), zap.Duration("merge_time", time.Since(t0)))
 
-	for filename := range b.fileList {
-		m.seenBlocks.Add(filename) // add them to 'seenbefore' right before deleting them on gs
-		uploaded = append(uploaded, filename)
+	for _, obf := range b.fileList {
+		m.seenBlocks.Add(obf.canonicalName) // add them to 'seenbefore' right before deleting them on gs
```

Comment:
What about clean up of unused filenames? I don't see where they would be deleted. I'm not familiar with the merger code so maybe it's already handled, raising the point since I did not see any modifications related to this...